### ISSUE TITLE
[YSQL][docs] Document creating indexes on jsonb columns

### ISSUE THREAD

**Issue by ndeodhar** [yugabyte/yugabyte-db#2118]
We support creating indexes on jsonb columns in YSQL. This should be added to our documentation:
https://docs.yugabyte.com/latest/api/ysql/datatypes/type_json/

```
postgres=# create table test(a int primary key, b jsonb);
CREATE TABLE
postgres=# create index on test((b->>'x'));
CREATE INDEX
postgres=# insert into test(a,b) values(1, '{"x":"xval1","y":"yval1"}');
INSERT 0 1
postgres=# insert into test(a,b) values(2, '{"x":"xval2","y":"yval2"}');
INSERT 0 1
postgres=# select * from test where b->>'x'='xval2';
 a |              b               
---+------------------------------
 2 | {"x": "xval2", "y": "yval2"}
(1 row)

postgres=# explain select * from test where b->>'x'='xval2';
                                QUERY PLAN                                 
---------------------------------------------------------------------------
 Index Scan using test_expr_idx on test  (cost=0.00..4.12 rows=1 width=36)
   Index Cond: ((b ->> 'x'::text) = 'xval2'::text)
(2 rows)
```

**Comment by Bessonov** [comment]
@ndeodhar CockroachDB take a little bit another approach for indexing. It's auto populating index with all jsonb fields:
https://www.cockroachlabs.com/docs/stable/inverted-indexes.html

**Comment by bllewell** [comment]
Fixed 19-Mar-2020 by [PR #4011](https://github.com/yugabyte/yugabyte-db/pull/4011). See [Create indexes and check constraints on JSON columns](https://docs.yugabyte.com/latest/api/ysql/datatypes/type_json/create-indexes-check-constraints/).
